source/base/function_parser.cc

```diff
@@ -309,48 +186,9 @@ FunctionParser<dim>::init_muparser() const
           // we may find after function names
           std::string transformed_expression = expressions[component];
 
-          const char *function_names[] = {// functions predefined by muparser
-                                          "sin",
-                                          "cos",
-                                          "tan",
-                                          "asin",
-                                          "acos",
-                                          "atan",
-                                          "sinh",
-                                          "cosh",
-                                          "tanh",
-                                          "asinh",
-                                          "acosh",
-                                          "atanh",
-                                          "atan2",
-                                          "log2",
-                                          "log10",
-                                          "log",
-                                          "ln",
-                                          "exp",
-                                          "sqrt",
-                                          "sign",
-                                          "rint",
-                                          "abs",
-                                          "min",
-                                          "max",
-                                          "sum",
-                                          "avg",
-                                          // functions we define ourselves above
-                                          "if",
-                                          "int",
-                                          "ceil",
-                                          "cot",
-                                          "csc",
-                                          "floor",
-                                          "sec",
-                                          "pow",
-                                          "erfc",
-                                          "rand",
-                                          "rand_seed"};
-          for (const auto &function_name_c_string : function_names)
+          for (const auto &current_function_name : internal::function_names)
             {
-              const std::string  function_name        = function_name_c_string;
+              const std::string  function_name(current_function_name);
```

Comment:
`clang-tidy`: You don't need the copy here. `current_function_name` is already of type `std::string`.

Reply:
Done.